src/System.Runtime/tests/System/DelegateTests.cs

```diff
@@ -46,6 +59,35 @@ public static void ClosedStaticDelegate()
             Assert.Same(foo.structField.o2, returnedStruct.o2);
         }
 
+        [Fact]
+        public static void ClosedStaticDelegateSerialization()
+        {
+            var t = new TestSerializableClass();
+            Assert.Equal(1, t.x);
+            Action d = t.IncrementX;
+            d();
+            Assert.Equal(2, t.x);
+
+            // Serialize the delegate
+            var binaryFormatter = new BinaryFormatter();
+            byte[] serializedBytes;
+            using (var memoryStream = new MemoryStream())
+            {
+                binaryFormatter.Serialize(memoryStream, d);
+                serializedBytes = memoryStream.ToArray();
+            }
+
+            // Deserialize the delegate
+            using (var memoryStream = new MemoryStream(serializedBytes))
+            {
+                d = (Action)binaryFormatter.Deserialize(memoryStream);
+            }
+            t = (TestSerializableClass)d.Target;
```

Comment:
We have a helper for all of this BinaryFormatter stuff:
https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Runtime/Serialization/Formatters/BinaryFormatterHelpers.cs#L13
so you could use that to remove most of this code.